geo/Cargo.toml

```diff
@@ -22,19 +22,20 @@ geographiclib-rs = { version = "0.2" }
 
 proj = { version = "0.20.3", optional = true }
 
-geo-types = { version = "0.6.2", path = "../geo-types", features = ["rstar"] }
+geo-types = { version = "0.6.2", optional = true, path = "../geo-types", features = ["rstar"] }
```

Comment:
I'm not 100% I understand the changes in this file. Surely geo cannot be built without geo-types, so making it an optional dependency seems fraught.

I'm guessing this is related to enabling the `relative_eq` by default only as a devDependency... Is that right? I see why that might be tricky.

A maybe radical alternative which avoids the problem is to get rid of the `relative_eq` feature you've introduced and simply have the new functionality you've written enabled by default. 

I appreciate the intent, because we try to keep geo-types svelte, but since geo-types already unconditionally relies on the `approx` crate, including your new code for the approx impls seems like an acceptable addition to me.

Anyone else care to weigh in? 
/cc @georust/geo-publishers

Reply:
Thanks for the review, this is a tricky one 

yes to a radical alternatives... this frustrates me ..Let me state the problem in very simplistic terms just so someone who has seen this before can say .. Oh that is easy.

I want different features to be present on (A) the release build and (B) the dev build.

Making geo-types optional is bad when it must always be included. I was constrained here ... there are long threads on stackoverflow about this -  and meandering discussions amongst rustlang  developers  with lots of well meaning comments along the line of .. we can do better but no action.  I hope I am just making a silly mistake.

Reply:
To be clear, I'm just waiting a few days to solicit feedback from @georust/geo-publishers

The specific question is:

Since we already include approx by default in geo-types, are we ok with adding these implementations of approx traits on geo-types by default as well.

Reply:
This looks like the feature we'd want: https://github.com/rust-lang/cargo/issues/7916

Reply:
> This looks like the feature we'd want: [rust-lang/cargo#7916](https://github.com/rust-lang/cargo/issues/7916)

And if so, it looks like this feature should stabilize soon: https://github.com/rust-lang/cargo/pull/8997

So it seems like we have a few options:

1. Wait to merge this until https://github.com/rust-lang/cargo/issues/7916 lands on stable Rust
2. Make geo-types optional, which is what this pull request currently does. When https://github.com/rust-lang/cargo/issues/7916 lands, we can make `geo-types` required again
3. Keep geo-types required, but always include the `relative_eq` feature. Even on non-test builds. When https://github.com/rust-lang/cargo/issues/7916 lands, we can turn on `relative_eq` only in `dev-dependencies`

I'm fine with any of these options, with a slight preference for option 3. It's unfortunate we introduce a new dependency for everyone, but it'll hopefully be temporary.

If we go with options 2 or 3, we should add a comment or open a GitHub Issue saying we can improve the situation when https://github.com/rust-lang/cargo/issues/7916 lands.

Reply:
I'm working on a slightly different proposal... hopefully I'll have something to share in 20min.

Reply:
please see https://github.com/martinfrances107/geo/pull/1

Reply:
What do you think of this solution @frewsxcv (et al)?

Currently in master:

- geo-types dependency on approx is not optional
- the methods in geo_types::private_utils are unconditionally built, but only used within geo-types if the `rstar` feature is enabled.

After this PR:

- geo-types dependency on approx is optional
- These new implementations for RelativeEq and AbsDiffEq are enabled with the now optional `approx` feature
- There is "public" API breakage, in that methods in geo_types::private_utils are now skipped unless the rstar feature which requires them is enabled (`geo` builds `geo-types` w/ `rstar` feature, so this won't affect `geo`).

So technically, if some other crate is using the methods in private_utils, and not enabling the `rstar` feature, this would break for them. Even though this technically breaks a public API, I _think_ it's OK to treat this as a minor change since the changes are in a module with the word "private" in its name. WDYT?

Reply:
SGTM! Great solution